update `getBindingsProxy` documentation (#12787)
cloudflare/cloudflare-docs: content/workers/wrangler/api.md

The `getBindingsProxy` function is used to get a proxy for **local** `workerd` bindings that can be then used in code running via Node.js processes for Workers and Pages projects.

> This is useful for emulating bindings in applications targeting Workers, but running outside the Workers runtime (e.g. framework local development servers running in Node.js), or for testing purposes (e.g. ensuring a function properly interacts with a binding).

Reviewer: ```suggestion
The `getBindingsProxy` function is useful for emulating bindings in applications targeting Workers, but running outside the Workers runtime (for example, framework local development servers running in Node.js), or for testing purposes (for example, ensuring a Function properly interacts with a binding).
```
Check chat questions

Reviewer: Updated!

{{<Aside type="note">}}

Binding proxies provided by this function are a best effort emulation of the real production bindings. Although they are designed to be as close as possible to the real thing, there might be slight differences and inconsistencies between the two.

{{</Aside>}}

### Syntax

```js
const bindingsProxy = await getBindingsProxy(options);
```

### Parameters

{{<definitions>}}

*   `options` {{<type>}}object{{</type>}} {{<prop-meta>}}optional{{</prop-meta>}}

    *   Optional options object containing preferences for the bindings:
        * `configPath` {{<type>}}string{{</type>}}

>           The path to the config object to use (default `wrangler.toml`)

Reviewer: ```suggestion
          The path to the configuration object to use (default `wrangler.toml`).
```

        * `experimentalJsonConfig` {{<type>}}boolean{{</type>}}

>           If `true`, allows the utility to read a JSON config file (e.g. `wrangler.json`).

Reviewer: ```suggestion
          If `true`, allows the utility to read a JSON config file (for example, `wrangler.json`).
```

        * `persist` {{<type>}}boolean | { path: string }{{</type>}}

>           Indicates if and where to persist the bindings data. If not present or `true`, defaults to the same location
>           used by Wrangler, so data can be shared between it and the caller. If `false`, no data is persisted to or read from the filesystem.

Reviewer: ```suggestion
          Indicates if and where to persist the bindings data. If not present or `true`, defaults to the same location used by Wrangler, so data can be shared between it and the caller. If `false`, no data is persisted to or read from the filesystem.
```

{{</definitions>}}

### Return Type

> `getBindingsProxy()` returns a `Promise` resolving to an object containing the following fields:

Reviewer: ```suggestion
`getBindingsProxy()` returns a `Promise` resolving to an object containing the following fields.
```

{{<definitions>}}

*   `bindings` {{<type>}}Record<string, unknown>{{</type>}}

>     *   Bindings proxies that can be used in the same way as production bindings. This matches the shape of the `env` object passed as the 2nd argument to modules-format workers. These proxy to binding implementations run inside `workerd`.

Reviewer: ```suggestion
    *   Bindings proxies that can be used in the same way as production bindings. This matches the shape of the `env` object passed as the second argument to modules-format workers. These proxy to binding implementations run inside `workerd`.
```
Is "modules-format workers" = Workers written in ES Modules format?

Author: > Is "modules-format workers" = Workers written in ES Modules format?
yes

{{</Aside>}}

### Syntax

```js
const bindingsProxy = await getBindingsProxy(options);
```

### Parameters

{{<definitions>}}

*   `options` {{<type>}}object{{</type>}} {{<prop-meta>}}optional{{</prop-meta>}}

    *   Optional options object containing preferences for the bindings, such are:
        * `configPath` {{<type>}}string{{</type>}}

          The path to the config object to use (default `wrangler.toml`)

        * `experimentalJsonConfig` {{<type>}}boolean{{</type>}}

          Flag requiring the utility to read a json config file (`wrangler.json`), if present, instead of the toml one (`wrangler.toml`)

        * `persist` {{<type>}}boolean | { path: string }{{</type>}}

          Indicates if and where to persist the bindings data, if not present or `true` it defaults to the same location
          used by wrangler v3: `.wrangler/state/v3` (so that the same data can be easily used by the caller and wrangler).

>           If `false` is specified no data is persisted on the filesystem.

Author: This looks good to me but I am afraid that `wrangler dev` can be a bit restrictive

I mentioned wrangler generically to:
 - include both `wrangler dev` _and_ `wrangler pages dev`
 - include all wrangler commands that can be used to populate bindings
 
 what do you think?

>           If `false` is specified no data is persisted on the filesystem.

Author: sure, sounds good 🙂👍

{{</definitions>}}

### Return Type

`getBindingsProxy()` returns a promise to an object containing the following fields:

{{<definitions>}}

*   `bindings` {{<type>}}Record<string, unknown>{{</type>}}

    *   The actual bindings proxies, such can be used exactly in the same manner production bindings would. They proxy to binding implementations run inside a `workerd` instance (run by Miniflare).
    *   Typescript Tip: `getBindingsProxy` is a generic function, you can specify to it the shape of the bindings record in order to have a more proper type without `unknown` values.

*   `caches` {{<type>}}object{{</type>}}

>     *   Emulation of the [workers `caches` runtime API](https://developers.cloudflare.com/workers/runtime-apis/cache/).

Author: 🤦

console.log(`MY_VARIABLE = ${bindings['MY_VARIABLE']}`);
```

This will print the following output: `MY_VARIABLE = test`.

### Supported bindings

All supported bindings found in your `wrangler.toml` are available to you via `bindings`.

The bindings supported by `getBindingsProxy` are:

 * [Environmental variables](/workers/wrangler/configuration/#environmental-variables)

 * [Service bindings](/workers/configuration/bindings/#service-bindings)

 * [KV namespace bindings](/workers/configuration/bindings/#kv-namespace-bindings)

 * [Durable Object bindings](/workers/configuration/bindings/#durable-object-bindings)

 * [R2 bucket bindings](/workers/configuration/bindings/#r2-bucket-bindings)

 * [Queue bindings](/workers/configuration/bindings/#queue-bindings)

 * [D1 database bindings](/workers/configuration/bindings/#d1-database-bindings)

 * [Workers AI bindings](/workers/configuration/bindings/#workers-ai-bindings)

>     * {{<render file="_ai-local-usage-charges.md" productFolder="workers">}}

Reviewer: ```suggestion
{{<render file="_ai-local-usage-charges.md" productFolder="workers">}}
```

>     * To use the `AI` binding, you'll need to set the `CLOUDFLARE_ACCOUNT_ID` and `CLOUDFLARE_API_TOKEN` environment variables to your Cloudflare [Account ID](/fundamentals/setup/find-account-and-zone-ids/) and a Workers AI enabled API token (refer to [Workers AI documentation](/workers-ai/get-started/rest-api/#1-get-an-api-token)) respectively.

Reviewer: ```suggestion
    * To use the `AI` binding, you need to set the `CLOUDFLARE_ACCOUNT_ID` and `CLOUDFLARE_API_TOKEN` environment variables to your Cloudflare [account ID](/fundamentals/setup/find-account-and-zone-ids/) and a [Workers AI enabled API token](/workers-ai/get-started/rest-api/#1-get-an-api-token) respectively.
```

console.log(`MY_VARIABLE = ${bindings['MY_VARIABLE']}`);
```

> which would print the following output: `MY_VARIABLE = test`

Reviewer: ```suggestion
This will print the following output: `MY_VARIABLE = test`.
```

### Supported bindings

All supported bindings found in your `wrangler.toml` are available to you via `bindings`.

The bindings supported by `getBindingsProxy` are:

 * [Environmental variables](/workers/wrangler/configuration/#environmental-variables)

 * [Service bindings](/workers/configuration/bindings/#service-bindings)

 * [KV namespace bindings](/workers/configuration/bindings/#kv-namespace-bindings)

 * [Durable Object bindings](/workers/configuration/bindings/#durable-object-bindings)

 * [R2 bucket bindings](/workers/configuration/bindings/#r2-bucket-bindings)

 * [Queue bindings](/workers/configuration/bindings/#queue-bindings)

 * [D1 database bindings](/workers/configuration/bindings/#d1-database-bindings)

 * [Workers AI bindings](/workers/configuration/bindings/#workers-ai-bindings)

    * __Important__: When using the AI binding locally, computation still happens on the Cloudflare network. Usage of this binding incurs usage charges even when running locally.

    * To use the `AI` binding, you'll need to set the `CLOUDFLARE_ACCOUNT_ID` and `CLOUDFLARE_API_TOKEN` environment variables to your

>      Cloudflare [Account ID](/fundamentals/setup/find-account-and-zone-ids/) and a Workers AI enabled API token (refer to [Workers AI documentation](/workers-ai/get-started/rest-api/#1-get-an-api-token)) respectively.

Reviewer: This needs to be a partial and it should also go in this section as well: https://developers.cloudflare.com/workers/wrangler/configuration/#ai

Author: I've converted the first bullet point into a partial and reused in both here and in the wrangler configuration page

The second bullet point is not applicable to the other page (only when using `getBindingsProxy` you need to set those environment variables) so I left that as is

I hope this works for you, please let me know 🙂 🙏
